src/lib/plot/xy-plot.js

```diff
@@ -225,11 +216,45 @@ class XYPlot extends React.Component {
     };
   }
 
-  render() {
-    const {width, height, animation} = this.props;
+  /**
+   * Checks if the plot is empty or not.
+   * Currently checks the data only.
+   * @returns {boolean} True for empty.
+   * @private
+   */
+  _isPlotEmpty() {
+    const {data} = this.state;
+    return !data || !data.length || ![].concat(...data).filter(d => d).length
```

Comment:
I don't think the `[].concat(...data)` is needed, unless `data` has a `length` attribute but doesn't have `.filter` which should break everything anyway.